extensions/wikia/AdEngine/js/lookup/rubiconFastlane.js

```diff
@@ -0,0 +1,175 @@
+/*global define*/
+define('ext.wikia.adEngine.lookup.rubiconFastlane', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.adTracker',
+	'ext.wikia.adEngine.utils.adLogicZoneParams',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, adTracker, adLogicZoneParams, doc, log, win) {
+	'use strict';
+
+	var called = false,
+		config = {
+			oasis: {
+				TOP_LEADERBOARD: [[728, 90],[970, 250]],
+				TOP_RIGHT_BOXAD: [[300, 250],[300, 600]],
+				LEFT_SKYSCRAPER_2: [[160,600]],
+				LEFT_SKYSCRAPER_3: [[160,600]],
+				INCONTENT_BOXAD_1: [[300, 250]],
+				PREFOOTER_LEFT_BOXAD: [[300, 250]],
+				PREFOOTER_RIGHT_BOXAD: [[300, 250]]
+			},
+			mercury: {
+				MOBILE_IN_CONTENT: [[300, 250]],
+				MOBILE_PREFOOTER: [[300, 250]],
+				MOBILE_TOP_LEADERBOARD: [[320, 50]]
+			}
+		},
+		logGroup = 'ext.wikia.adEngine.lookup.rubiconFastlane',
+		priceMap = {},
+		response = false,
+		rubiconSlots = [],
+		slots = {},
+		timing;
+
+	function getSlots(skin) {
+		var context = adContext.getContext(),
+			pageType = context.targeting.pageType,
+			slotName;
+
+		slots = config[skin];
+		if (skin === 'oasis' && pageType === 'home') {
```

Comment:
It'll make additional requests for each size per slot so we would like to prevent it. Also the same logic is in oxBidder.js so it may be extracted during work on ADEN-2576.